### PR TITLE
NEX-8/Get-user-profile

### LINKED ISSUE #8: [NEX-8] Get user profile

### PR DESCRIPTION
**Pull request by LuckyOx04**

### CODE REVIEW

File: nexus/src/main/java/com/example/nexus/model/payload/response/ProfileInfoResponse.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.example.nexus.model.payload.response;
+
+public record ProfileInfoResponse(
```
Comment (stodorichkov) [inline nexus/src/main/java/com/example/nexus/model/payload/response/ProfileInfoResponse.java]: Just ProfileResponse is ok

File: nexus/src/main/java/com/example/nexus/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/user")
+@RequiredArgsConstructor
+public class UserController {
+    private final UserService userService;
+
+    @GetMapping("/info")
```
Comment (stodorichkov) [inline nexus/src/main/java/com/example/nexus/controller/UserController.java]: move it in ProfileController and use @GetMapping()

File: nexus/src/main/java/com/example/nexus/service/UserService.java [hunk 3]
```diff
@@ -9,4 +11,5 @@ public interface UserService {
     Page<UserResponse> getUsers(Pageable pageable);
     void addUserRole(String username);
     void removeUserRole(String username);
+    ProfileInfoResponse getProfileInfo(HttpServletRequest request);
```
Comment (stodorichkov) [inline nexus/src/main/java/com/example/nexus/service/UserService.java]: Move it in ProfileService

File: nexus/src/test/java/com/example/nexus/service/FileServiceImplTests.java [hunk 4]
```diff
@@ -74,7 +74,6 @@ void upload_validFile_expectUpload() {
             );
 
             when(this.imageConfig.getDir()).thenReturn("/path/to/images");
-            when(this.imageConfig.getBaseUrl()).thenReturn(baseUrl);
```
Comment (stodorichkov) [inline nexus/src/test/java/com/example/nexus/service/FileServiceImplTests.java]: Why remove it

File: nexus/src/main/java/com/example/nexus/service/UserServiceImpl.java [hunk 5]
```diff
@@ -88,4 +93,20 @@ public void removeUserRole(String username) {
 
         userRepository.save(user);
     }
+
+    @Override
+    public ProfileInfoResponse getProfileInfo(HttpServletRequest request) {
```
Comment (stodorichkov) [inline nexus/src/main/java/com/example/nexus/service/UserServiceImpl.java]: Move in ProfileServiceImpl

File: nexus/src/test/java/com/example/nexus/service/ProfileServiceImplTests.java [hunk 6]
```diff
@@ -16,7 +16,6 @@
 @ExtendWith(MockitoExtension.class)
 class ProfileServiceImplTests {
     private static Profile profile;
-    private static AddMoneyRequest addMoneyRequest;
```
Comment (stodorichkov) [inline nexus/src/test/java/com/example/nexus/service/ProfileServiceImplTests.java]: why remove it

File: nexus/src/test/java/com/example/nexus/service/UserServiceImplTests.java [hunk 7]
```diff
@@ -186,14 +206,33 @@ public void removeUSerRole_everythingIsFine_expectToWork() {
...
+        this.userService.removeUserRole("Username");
 
         user.getRoles().remove(newRole);
 
         verify(this.userRepository).save(userCaptor.capture());
         assertEquals(user, userCaptor.getValue());
     }
+
+    @Test
+    public void getUsernameFromAuthRequest_profileNotFound_expectNotFoundException() {
```
Comment (stodorichkov) [inline nexus/src/test/java/com/example/nexus/service/UserServiceImplTests.java]: Move in ProfileServiceImplTests

**Review by stodorichkov — CHANGES_REQUESTED** [review]

File: nexus/src/test/java/com/example/nexus/service/ProfileServiceImplTests.java (hunk 6 above)
Comment (LuckyOx04) [reply nexus/src/test/java/com/example/nexus/service/ProfileServiceImplTests.java]: This created a warning that this declaration should be local and not global, so it was moved in the addMoney test.

**Review by stodorichkov — APPROVED** [review]

**Pull request merged**